workflow/workflow_commands.go

```diff
@@ -1445,6 +1448,76 @@ func TraceWorkflow(c *cli.Context) error {
 	return nil
 }
 
+func UpdateWorkflow(c *cli.Context) error {
+	wid := c.String(common.FlagWorkflowID)
+	rid := c.String(common.FlagRunID)
+	name := c.String(common.FlagName)
+	uid := c.String(common.FlagUpdateID)
+	first_execution_run_id := c.String(common.FlagUpdateFirstExecutionRunID)
+	args, err := common.UnmarshalInputsFromCLI(c)
+	if err != nil {
+		return err
+	}
+	waitPolicy, ok := updateWaitPolicyMap[c.String(common.FlagUpdateWaitPolicy)]
+	if !ok {
+		return fmt.Errorf("must specify valid wait policy: %v", strings.Join(mapKeysToArray(updateWaitPolicyMap), ", "))
+	}
+	if (len(name) == 0 && len(uid) == 0) || (len(name) > 0 && len(uid) > 0) {
+		return fmt.Errorf("either %v or %v should be provided", color.Yellow(c, "--%v", common.FlagName), color.Yellow(c, "--%v", common.FlagUpdateID))
+	}
+
+	if len(name) > 0 {
+		request := sdkclient.UpdateWorkflowWithOptionsRequest{
+			WorkflowID:          wid,
+			RunID:               rid,
+			UpdateName:          name,
+			Args:                args,
+			FirstExecutionRunID: first_execution_run_id,
+			// TODO: uncomment when waitPolicy is available
+			// waitPolicy:          waitPolicy,
+		}
+		return updateWorkflowHelper(c, &request, waitPolicy)
+	} else {
+		// TODO: implement this when GetWorkflowUpdateHandle is available in SDKClient
+		return fmt.Errorf("Getting update result is not yet implemented")
+	}
+
+}
+
+// TODO: remove waitPolicy after it is available in the request
+func updateWorkflowHelper(c *cli.Context, request *sdkclient.UpdateWorkflowWithOptionsRequest, waitPolicy interface{}) error {
+	ctx, cancel := common.NewContext(c)
+	defer cancel()
+	sdk, err := client.GetSDKClient(c)
+	if err != nil {
+		return err
+	}
+
+	workflowUpdateHandle, err := sdk.UpdateWorkflowWithOptions(ctx, request)
+
+	if err != nil {
+		return fmt.Errorf("unable to update workflow: %w", err)
+	}
+
+	switch waitPolicy {
+	case enumspb.UPDATE_WORKFLOW_EXECUTION_LIFECYCLE_STAGE_COMPLETED:
+		var valuePtr interface{}
+		err = workflowUpdateHandle.Get(ctx, &valuePtr)
+		if err != nil {
+			return fmt.Errorf("unable to update workflow: %w", err)
```

Comment:
Can you use a different error message than on line 1499? It makes debugging easier